docs/spec_draft.md

```diff
@@ -1798,6 +1799,92 @@ tensor and produces a `result` tensor.
+| `has_side_effect`       | constant of type `i1`                                               |
+| `backend_config`        | constant of type `string`                                           |
+| `api_version`           | enum of `API_VERSION_ORIGINAL`, `API_VERSION_STATUS_RETURNING`,     |
+|                         | and `API_VERSION_STATUS_RETURNING_UNIFIED`                          |
+| `called_computations`   | variadic number of `function` references                            |
+| `operand_layouts`       | variadic number of 1-dimensional tensor constants of type `si64`    |
+| `result_layouts`        | variadic number of 1-dimensional tensor constants of type `si64`    |
+| `output_tuple_indices`  | 1-dimensional tensor constant of type `si64`                        |
+| `operand_index`         | constant of type `si64`                                             |
+| `operand_tuple_indices` | 1-dimensional tensor constant of type `si64`                        |
+
+### Outputs
+
+| Name      | Type                                                                |
+|-----------|---------------------------------------------------------------------|
+| `results` | variadic number of tensors of any supported type, tokens, or tuples |
+
+### Constraints
+
+  * (C1) `operand_layouts` & `result_layouts` attributes can be specified under
+          the following constraints:
+    * Either both `operand_layouts` and `result_layouts` are specified or none.
+    * None of the `inputs` are of tuple type.
+    * None of the `results` are of tuple type except the common case of single
+      tuple result packing non-tuple values is allowed.
+    * `operand_layouts[i]` = `layout(inputs[i])`, for all `i`.
+    * `result_layouts[i]` = `layout(results[i])`, for all `i`.
+    * Only tensor types can have non-empty layout.
+    * Layout of an `N`-ranked tensor must be a permutation of `[0, N)`.
+  * (C2) `output_operand_aliases` can be specified under the following
+         constraints:
+    * 0 $\le$ `operand_index` $\lt$ `size(inputs)`.
+    * `type(inputs[operand_index])` = `tuple`.
+    * `shape(inputs[operand_index][i0][i1]..[iN-1])` =
+      `shape(results[j0][j1]..[jM-1]])`, where,
+      `i0, i1, ..` = `operand_tuple_indices[0], operand_tuple_indices[1], ..`,
+      `j0, j1, ..` = `operand_tuple_indices[0], operand_tuple_indices[1], ..`,
+      `size(operand_tuple_indices) = N`, and `size(output_tuple_indices) = M`.
+
+### Examples
+
+```mlir
+%results = "stablehlo.custom_call"(%inputs0) {
+  call_target_name = "foo",
+  has_side_effect = false,
+  backend_config = "bar",
+  api_version = 1 : i32,
+  called_computations = [@foo],
+  operand_layouts = [dense<> : tensor<0xindex>],
+  output_operand_aliases = [
+    #stablehlo.output_operand_alias<output_tuple_indices = [],
+                                    operand_index = 0,
+                                    operand_tuple_indices = []>],
+  result_layouts = [dense<> : tensor<0xindex>]
```

Comment:
Nit: Following the order of Inputs, `result_layouts` should come after `operand_layouts` (here and ODS)?